lib/puppet/functions/dnsquery/soa.rb

```diff
@@ -1,23 +1,19 @@
 # frozen_string_literal: true
 
-# Retrieves DNS TXT records and returns it as an array. Each record in the
-# array will be a array containing the strings of the TXT record.
-#
-# An optional lambda can be given to return a default value in case the
-# lookup fails. The lambda will only be called if the lookup failed.
+# Retrieves DNS SOA records and returns it as a hash.
 Puppet::Functions.create_function(:'dnsquery::soa') do
+  # @param question the dns question to lookup
+  # @param block an optional lambda to return a default value in case the lookup fails
+  # @return The SOA record matching domain
   dispatch :dns_soa do
-    param 'String', :record
+    param 'String', :question
+    optional_block_param :block
+    return_type 'Hash[String, String, 7]'
```

Comment:
You could go as far as defining a `Struct`, but that's probably too far.

Reply:
done but means this pr is growing quite a bit :/

Reply:
Yes, so it may be fine to leave it at a simple Hash.